test/e2e/clusterclass_changes.go

```diff
@@ -0,0 +1,403 @@
+	Expect(ctx).NotTo(BeNil(), "ctx is required for ModifyMachineDeploymentViaClusterClassAndWait")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling ModifyMachineDeploymentViaClusterClassAndWait")
+	Expect(input.ClusterClass).ToNot(BeNil(), "Invalid argument. input.ClusterClass can't be nil when calling ModifyMachineDeploymentViaClusterClassAndWait")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling ModifyMachineDeploymentViaClusterClassAndWait")
+
+	mgmtClient := input.ClusterProxy.GetClient()
+
+	for _, mdClass := range input.ClusterClass.Spec.Workers.MachineDeployments {
+		// Continue if the MachineDeploymentClass is not using a BootstrapConfigTemplate.
+		if mdClass.Template.Bootstrap.Ref == nil {
+			continue
+		}
+
+		log.Logf("Modifying the BootstrapConfigTemplate of MachineDeploymentClass %q of ClusterClass %s/%s", mdClass.Class, input.ClusterClass.Namespace, input.ClusterClass.Name)
+
+		// Retrieve BootstrapConfigTemplate object.
+		bootstrapConfigTemplateRef := mdClass.Template.Bootstrap.Ref
+		bootstrapConfigTemplate, err := external.Get(ctx, mgmtClient, bootstrapConfigTemplateRef, input.Cluster.Namespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Create a new BootstrapConfigTemplate object with a new name and ModifyBootstrapConfigTemplateFields set.
+		newBootstrapConfigTemplate := bootstrapConfigTemplate.DeepCopy()
+		newBootstrapConfigTemplateName := fmt.Sprintf("%s-%s", bootstrapConfigTemplateRef.Name, util.RandomString(6))
+		newBootstrapConfigTemplate.SetName(newBootstrapConfigTemplateName)
+		newBootstrapConfigTemplate.SetResourceVersion("")
+		for fieldPath, value := range input.ModifyBootstrapConfigTemplateFields {
+			Expect(unstructured.SetNestedField(newBootstrapConfigTemplate.Object, value, strings.Split(fieldPath, ".")...)).To(Succeed())
+		}
+		Expect(mgmtClient.Create(ctx, newBootstrapConfigTemplate)).To(Succeed())
+
+		// Patch the BootstrapConfigTemplate ref of the MachineDeploymentClass to reference the new BootstrapConfigTemplate.
+		patchHelper, err := patch.NewHelper(input.ClusterClass, mgmtClient)
+		Expect(err).ToNot(HaveOccurred())
+		bootstrapConfigTemplateRef.Name = newBootstrapConfigTemplateName
+		Expect(patchHelper.Patch(ctx, input.ClusterClass)).To(Succeed())
+
+		log.Logf("Waiting for MachineDeployment rollout for MachineDeploymentClass %q to complete.", mdClass.Class)
+		for _, mdTopology := range input.Cluster.Spec.Topology.Workers.MachineDeployments {
+			// Continue if the MachineDeploymentTopology belongs to another MachineDeploymentClass.
+			if mdTopology.Class != mdClass.Class {
+				continue
+			}
+
+			log.Logf("Waiting for MachineDeployment rollout for MachineDeploymentTopology %q (class %q) to complete.", mdTopology.Name, mdTopology.Class)
+			Eventually(func() error {
```

Comment:
Same as above (wait for machines to actually rollout)

Reply:
Similar argument but not the same:

My idea was to just check that it's rolled out to the MD because the MD rollout is the concern of the MD controller (and we have other e2e tests covering this)

It also speeds up the test a bit because we don't have to wait that long.

It also helps to focus this test on the ClusterClass functionality.